Classes/Middleware/SetAllowCookieMiddleware.php

```diff
@@ -0,0 +1,71 @@
+<?php
+declare(strict_types=1);
+
+namespace Netlogix\Varnish\AllowCookie\Middleware;
+
+use Neos\Flow\Annotations as Flow;
+use Netlogix\Varnish\AllowCookie\Exception\InvalidPathPatternGiven;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Throwable;
+
+/**
+ * @Flow\Proxy(false)
+ */
+final class SetAllowCookieMiddleware implements MiddlewareInterface
+{
+
+    private const SET_COOKIE = 'Set-Cookie';
+    private const X_ALLOW_COOKIE = 'X-Allow-Cookie';
+
+    private array $allowedRequestPathPatterns = [];
+
+    public function injectSettings(array $settings): void
```

Comment:
Flow has no DI on Proxy(false) classes, this should not work. That's not covered by any test because those are only unit tests calling the inject method manually.

Reply:
You are totally right. I removed the Proxy annotation and got rid of the final to avoid confusion (final is removed by Flow when building proxy classes)